docs/UpdatingMajorVersionAndTFM.md

```diff
@@ -0,0 +1,37 @@
+# Updating to a new Major Version & TFM
+
+At the end of each release cycle, we update our Major Version & TFM (Target Framework Moniker) in `main`, in preparation for the next major release. This doc describes the process of doing those updates, which can include many subtle gotchas in the aspnetcore repo.
+
+## Updating Major Version
+
+Typically, we will update the Major Version before updating the TFM. This is because updating Major Version only requires a logical agreement that `main` is no longer the place for work on the previous major release, while updating the TFM requires waiting for dotnet/runtime to update their TFM so that we can ingest that change. For an example, [this](https://github.com/dotnet/aspnetcore/pull/35402) is the PR where we updated our branding from 6.0.0 to 7.0.0.
```

Comment:
Some questions to help me understand this a little bit more:

- I assume because of the TFM updates requirement on dotnet/runtime, TFM updates can only happen in a dependency update PR. AKA it's part of the process of ingesting an update that goes from 6.x.x to 7.x.x.

- What does the `Major Version` impact? Is that just what gets set in local dev builds as the package version? So `7.0.0-dev` or `6.0.0-dev`.

Reply:
Yeah, in a dependency update PR. MajorVersion just updates the branding of the assets we produce in all builds, local & official